src/index.js

```diff
@@ -211,46 +28,32 @@ function addMetadata({ types: t }) {
     name: 'addMetadata',
     visitor: {
       Program(babelPath, state) {
-        const GET_TEST_METADATA = 'getTestMetadata';
+        if (!state.opts.enabled) return;
+
         const { filename } = state.file.opts;
         state.opts.shouldLoadFile = shouldLoadFile(filename);
 
-        if (!state.opts.shouldLoadFile) {
-          return;
-        }
+        if (!state.opts.shouldLoadFile) return;
 
-        state.opts.transformedModules = [];
-        state.opts.setupCall;
-        state.opts.moduleFunction;
-        state.opts.hooksIdentifier;
-        state.opts.getTestMetadataUID =
-          babelPath.scope.generateUidIdentifier(GET_TEST_METADATA);
+        const metadata = babelPath.get('body')
+          .filter((n) => n.type === 'ImportDeclaration')
+          .find(n => {
+            return n.get('specifiers').some(s => s.get('container').container.imported.name === "getTestMetadata");
```

Comment:
Can we use `getNodeProperty` here? Ran into an error getting 'name' from undefined.
```
            return n
              .get('specifiers')
              .some(
                (s) =>
                  getNodeProperty(
                    s.get('container'),
                    'container.imported.name'
                  ) === 'getTestMetadata'
              );
```